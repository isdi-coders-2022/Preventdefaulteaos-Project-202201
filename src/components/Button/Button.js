import {
  ButtonCreate,
  ButtonFilter,
  ButtonOnCard,
  ButtonWood,
} from "../../jsStyles/buttonStyles";
import PropTypes from "prop-types";

const Button = ({ className, actionOnClick, text, type }) => {
  return (
    <>
      {type === "wood" && (
        <ButtonWood className={className} onClick={actionOnClick}>
          {text}
        </ButtonWood>
      )}
      {type === "onCard" && (
        <ButtonOnCard className={className} onClick={actionOnClick}>
          {text}
        </ButtonOnCard>
      )}
      {type === "filter" && (
        <ButtonFilter className={className} onClick={actionOnClick}>
          {text}
        </ButtonFilter>
      )}
      {type === "create" && (
        <ButtonCreate className={className} onClick={actionOnClick}>
          {text}
        </ButtonCreate>
      )}
    </>
  );
};

Button.propTypes = {
  className: PropTypes.string,
  actionOnClick: PropTypes.func,
  text: PropTypes.string,
  type: PropTypes.string,
};

export default Button;
