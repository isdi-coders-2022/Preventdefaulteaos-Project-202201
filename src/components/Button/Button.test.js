import { render, screen } from "@testing-library/react";
import Button from "./Button";

describe("Given a Button component", () => {
  describe("When it's instanciated with 'create' filter", () => {
    test("Then it should not render a button", () => {
      const buttonFilter = "create";

      // eslint-disable-next-line no-unused-vars
      const view = render(<Button type={buttonFilter} />);

      const buttonRender = screen.getByRole("button");

      expect(buttonRender).toBeInTheDocument();
    });
  });
});
