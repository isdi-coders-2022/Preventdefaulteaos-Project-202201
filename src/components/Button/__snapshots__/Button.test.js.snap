// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Button component When executing the component Then it should render 1`] = `
<button
  className="sc-bdvvtL sc-eCImPb ewqzMe iKKLDN"
/>
`;
